libs/Meact/useState.ts

```diff
@@ -0,0 +1,30 @@
+import { rendering } from "../../src/main";
+
+type UseStateReturn<T> = [T, (newState: T) => void];
+
+type CurrentState<T> = {
+  value: T;
+  setState: (newState: T) => void;
+};
+
+let stateIndex = 0;
+const stateArr: CurrentState<unknown>[] = [];
+export const useState = <T extends {}>(initState: T): UseStateReturn<T> => {
+  if (stateArr.length === stateIndex) {
+    const state = {
+      value: initState,
+      setState(newValue) {
+        state.value = newValue;
+        stateIndex = 0;
+        rendering();
+      },
+    };
+
+    stateArr.push(state);
+  }
```

Comment:
p3: 현재는 함수를 직접 할당하다보니 ,불변을 보장하기 힘들 것 같아요. 아래 예시 느낌으로 재 선언하는 방식으로 변경주셔도 좋을 것 같습니다

```suggestion
  if (stateArr.length === stateIndex) {
    const state = {
      value: initState,
      setState(newValue) {
        const currentState = stateArr[stateIndex];
        stateArr[stateIndex] = { ...currentState, value: newValue }; // 불변성 유지
        rendering();
      },
    };
    stateArr.push(state);
  }

```

Reply:
반영하였습니다!